### PR TITLE
Introduce the 'aap_rules_validation' role

### PR DESCRIPTION
**Pull request by w4hf**
# What does this PR do?
Introduce a new role : `aap_rules_validation`
An ansible role which audit the declared AAP configuration and validate it against a set of user-defined rules. 

# How should this be tested?
Two test playbooks are delivered with the role under the folder `tests/playbooks` :
* `aap_rules_validation_filetree_read.yml`
* `aap_rules_validation_include.yml`
A set of example rules to test against are available under `tests/configs/controller_rules.yml`

# Is there a relevant Issue open for this?
n/a

# Other Relevant info, PRs, etc
I demoed this role during the community of practice meeting under the controller_policy name which was rectified to adapt to the AAP strategy.

### CODE REVIEW

**Comment by djdanielsson** [comment]
I wonder if the name shouldn't have controller in the name if the plan is in the future to support all the components

File: roles/aap_rules_validation/tasks/check_credentials_encryption.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+---
+- name: Init sensitive data list
+  ansible.builtin.set_fact:
+    __sensitive_data: []
+    __unencrypted: []
+
+- name: Set sensitive list
+  ansible.builtin.set_fact:
+    __sensitive_data: "{{ __sensitive_data + ([item.key] | product(item.value)) }}"
+  with_dict: "{{ rule['credential_sensitive_data'] }}"
```
Comment (ivarmu) [inline roles/aap_rules_validation/tasks/check_credentials_encryption.yml]: The `with_dict` is considered legacy. Should use `loop: "{{ rule['credential_sensitive_data'] | dict2items }}"` instead.

See https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_loops.html#with-dict

File: roles/aap_rules_validation/tasks/check_roles.yml [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+- name: Verify forbidden roles presence
+  when: rule['forbidden_roles'] is defined
+        and rule['forbidden_roles'] != None
+        and rule['forbidden_roles'] | length > 0
+  block:
+
+    - name: Set forbidden roles list
+      ansible.builtin.set_fact:
+        __forbidden_roles: "{{ __forbidden_roles + ([item.key] | product(item.value)) }}"
+      with_dict: "{{ rule['forbidden_roles'] }}"
```
Comment (ivarmu) [inline roles/aap_rules_validation/tasks/check_roles.yml]: The `with_dict` is considered legacy. Should use `loop: "{{ rule['forbidden_roles'] | dict2items }}"` instead.

See https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_loops.html#with-dict

**Review by ivarmu — CHANGES_REQUESTED** [review]
Amazing work with this! This will be very useful for many people for sure!

I only have to ask for two amends and the addition of a changelog fragment.

File: roles/aap_rules_validation/tasks/check_roles.yml (hunk 2 above)
Comment (w4hf) [reply roles/aap_rules_validation/tasks/check_roles.yml]: thanks !
done in [980ea8a3](https://github.com/redhat-cop/aap_configuration_extended/pull/16/commits/980ea8a3d4b56e441ef635ab4d623bce0b59ad3d)

File: roles/aap_rules_validation/tasks/check_credentials_encryption.yml (hunk 1 above)
Comment (w4hf) [reply roles/aap_rules_validation/tasks/check_credentials_encryption.yml]: thanks !
done in [980ea8a3](https://github.com/redhat-cop/aap_configuration_extended/pull/16/commits/980ea8a3d4b56e441ef635ab4d623bce0b59ad3d)

**Review by ivarmu — APPROVED** [review]
LGTM

**Review by djdanielsson — CHANGES_REQUESTED** [review]
I can merge once you fix the linting/pre-commit errors (the current ansible ones are expected so you can ignore those)

**Review by djdanielsson — APPROVED** [review]
LGTM

**Pull request merged**